cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -0,0 +1,302 @@
+
+    DatasetInstanceConfiguration creationProperties = new DatasetInstanceConfiguration(existing.getType(), properties);
+    enableExplore(instance, creationProperties);
+
+    //caling admin upgrade, after updating specification
+    executeAdmin(instance, "upgrade");
+  }
+
+  public void drop(Id.DatasetInstance instance) throws Exception {
+    LOG.info("Deleting dataset {}.{}", instance.getNamespaceId(), instance.getId());
+    DatasetSpecification spec = instanceManager.get(instance);
+    if (spec == null) {
+      throw new DatasetNotFoundException(instance);
+    }
+
+    dropDataset(instance, spec);
+  }
+
+  /**
+   * Executes an admin operation on a dataset instance.
+   *
+   * @param instance the instance to execute the admin operation on
+   * @param method the type of admin operation to execute
+   * @return the {@link DatasetAdminOpResponse} from the HTTP handler
+   * @throws Exception if something went wrong
+   */
+  public DatasetAdminOpResponse executeAdmin(Id.DatasetInstance instance, String method) throws Exception {
+    Object result = null;
+
+    // NOTE: one cannot directly call create and drop, instead this should be called thru
+    //       POST/DELETE @ /data/datasets/{instance-id}. Because we must create/drop metadata for these at same time
+    if (method.equals("exists")) {
+      result = opExecutorClient.exists(instance);
+    } else if (method.equals("truncate")) {
+      opExecutorClient.truncate(instance);
+    } else if (method.equals("upgrade")) {
+      opExecutorClient.upgrade(instance);
+    } else {
+      throw new HandlerException(HttpResponseStatus.NOT_FOUND, "Invalid admin operation: " + method);
+    }
+
+    return new DatasetAdminOpResponse(result, null);
+  }
+
+  /**
+   * Finds the {@link DatasetTypeMeta} for the specified dataset type name.
+   * Search order - first in the specified namespace, then in the 'system' namespace from defaultModules
+   *
+   * @param namespaceId {@link Id.Namespace} for the specified namespace
+   * @param typeName the name of the dataset type to search
+   * @return {@link DatasetTypeMeta} for the type if found in either the specified namespace or in the system namespace,
+   * null otherwise.
+   * TODO: This may need to move to a util class eventually
+   */
+  @Nullable
+  private DatasetTypeMeta getTypeInfo(Id.Namespace namespaceId, String typeName) {
+    Id.DatasetType datasetTypeId = Id.DatasetType.from(namespaceId, typeName);
+    DatasetTypeMeta typeMeta = implManager.getTypeInfo(datasetTypeId);
+    if (typeMeta == null) {
+      // Type not found in the instance's namespace. Now try finding it in the system namespace
+      Id.DatasetType systemDatasetTypeId = Id.DatasetType.from(Constants.SYSTEM_NAMESPACE_ID, typeName);
+      typeMeta = implManager.getTypeInfo(systemDatasetTypeId);
+    }
+    return typeMeta;
+  }
+
+  /**
+   * Drops a dataset.
+   * @param spec specification of dataset to be dropped.
```

Comment:
Insert a blank line before the @param, and remove the trailing period on the parameter.